test/004-article.test.js

```diff
@@ -266,17 +254,50 @@ describe('ARTICLE TEST SUITE', () => {
     it('should get an article', async () => {
       const response = await chai.request(server)
         .get(`${baseUrl}/${firstArticleSlug}`);
-
       expect(response.status).to.equal(200);
       expect(response.body.slug).to.equal('halt-and-catch-fire-davidhook');
     });
 
     it('should not get a non existent article', async () => {
       const response = await chai.request(server)
         .get('/api/v1/articles/non-existing-article');
+      expect(response.status).to.equal(404);
+      expect(response.body.error).to.equal('Article provided does not exist');
+    });
 
+    it('should not get a non existent article', async () => {
+      const response = await chai.request(server)
+        .get('/api/v1/articles/non-existing-article');
       expect(response.status).to.equal(404);
       expect(response.body.error).to.equal('Article provided does not exist');
     });
+
+    it('should have amount of returned data as specified', async () => {
+      const response = await chai.request(server)
+        .get('/api/v1/articles/?limit=2');
+      expect(response.status).to.equal(200);
+      expect(response.body.count).to.equal(2);
+    });
+
+    it('should return first page if query.page is less than 1', async () => {
+      const response = await chai.request(server)
+        .get('/api/v1/articles/?limit=2&page=-3');
+      expect(response.status).to.equal(200);
+      expect(response.body.page).to.equal(1);
+    });
+
+    it('should return first page if query.page is not provided', async () => {
+      const response = await chai.request(server)
+        .get('/api/v1/articles/?limit=2');
+      expect(response.status).to.equal(200);
+      expect(response.body.page).to.equal(1);
+    });
+
+    it('should return all articles if limit is not provided', async () => {
```

Comment:
To make this spec an accurate show of the fact that all articles are returned, I think using `page=2`would be great and then expect the count on that page to be 0 or something like that.